src/BuildOnWindows.cmake

```diff
@@ -160,96 +131,96 @@ elseif(BUILD_SPLIT_KERNEL_LIB)
   # Decouple with PyTorch cmake definition.
   install(TARGETS ${sycl_foreach_lib} DESTINATION "${TORCH_INSTALL_LIB_DIR}")
 
-  # Tensor kernel lib
-  set(sycl_tensor_lib torch_xpu_ops_sycl_tensor_kernels)
-  sycl_add_library(
-    ${sycl_tensor_lib}
-    SHARED
-    SYCL_SOURCES ${ATen_XPU_SYCL_TENSOR_SRCS})
-  target_compile_definitions(${sycl_tensor_lib} PRIVATE TORCH_XPU_BUILD_MAIN_LIB)
-  target_link_libraries(torch_xpu_ops_aten PUBLIC ${sycl_tensor_lib})
-  target_link_libraries(${sycl_tensor_lib} PUBLIC torch_xpu)
-  list(APPEND TORCH_XPU_OPS_LIBRARIES ${sycl_tensor_lib})
-
-  # Decouple with PyTorch cmake definition.
-  install(TARGETS ${sycl_tensor_lib} DESTINATION "${TORCH_INSTALL_LIB_DIR}")
-
-  # Norm and Loss kernel lib
-  set(sycl_norm_loss_lib torch_xpu_ops_sycl_norm_loss_kernels)
+  
+  # Other kernel lib
+  set(sycl_lib torch_xpu_ops_sycl_kernels)
   sycl_add_library(
-    ${sycl_norm_loss_lib}
+    ${sycl_lib}
     SHARED
-    SYCL_SOURCES ${ATen_XPU_SYCL_NORM_LOSS_SRCS})
-  target_compile_definitions(${sycl_norm_loss_lib} PRIVATE TORCH_XPU_BUILD_MAIN_LIB)
-  target_link_libraries(torch_xpu_ops_aten PUBLIC ${sycl_norm_loss_lib})
-  target_link_libraries(${sycl_norm_loss_lib} PUBLIC torch_xpu)
-  list(APPEND TORCH_XPU_OPS_LIBRARIES ${sycl_norm_loss_lib})
+    SYCL_SOURCES ${ATen_XPU_SYCL_OTHERS_SRCS})
+  target_compile_definitions(${sycl_lib} PRIVATE TORCH_XPU_BUILD_MAIN_LIB)
+  target_link_libraries(torch_xpu_ops_aten PUBLIC ${sycl_lib})
+  target_link_libraries(${sycl_lib} PUBLIC torch_xpu)
+  list(APPEND TORCH_XPU_OPS_LIBRARIES ${sycl_lib})
 
   # Decouple with PyTorch cmake definition.
-  install(TARGETS ${sycl_norm_loss_lib} DESTINATION "${TORCH_INSTALL_LIB_DIR}")
-
-  # Polynomial kernel lib
-  set(sycl_poly_lib torch_xpu_ops_sycl_poly_kernels)
-  sycl_add_library(
-    ${sycl_poly_lib}
-    SHARED
-    SYCL_SOURCES ${ATen_XPU_SYCL_POLY_SRCS})
-  target_compile_definitions(${sycl_poly_lib} PRIVATE TORCH_XPU_BUILD_MAIN_LIB)
-  target_link_libraries(torch_xpu_ops_aten PUBLIC ${sycl_poly_lib})
-  target_link_libraries(${sycl_poly_lib} PUBLIC torch_xpu)
-  list(APPEND TORCH_XPU_OPS_LIBRARIES ${sycl_poly_lib})
+  install(TARGETS ${sycl_lib} DESTINATION "${TORCH_INSTALL_LIB_DIR}")
 
-  # Decouple with PyTorch cmake definition.
-  install(TARGETS ${sycl_poly_lib} DESTINATION "${TORCH_INSTALL_LIB_DIR}")
+  list(APPEND TORCH_XPU_OPS_LIBRARIES torch_xpu_ops)
+  list(APPEND TORCH_XPU_OPS_LIBRARIES torch_xpu_ops_aten)
+else()
+  add_library(
```

Comment:
Please add comments why we need split objects into two separate archive libraries on Windows.